src/components/SearchPage.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState, useEffect, useContext, FormEventHandler } from 'react'
+import Card from './Card'
+import Pagination from './Pagination'
+import { fetchCards, debounce } from 'utils'
+
+const SearchPage = () => {
```

Comment:
something here isn't functioning correctly. Whenever you type a search term, it'll wait until you stop typing to hit the API, but it will hit the API for each combination you had typed.